### PR TITLE
Component for info alerts

### PR DESCRIPTION
**Pull request by Anxhela21**
**Related Issue:**  open-cluster-management/backlog#12024

### Description of changes
- generated ts definition for message query
- added "HeaderWithNotification" component

### CODE REVIEW

File: frontend/src/routes/SearchPage/components/MessageComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    //using translations:
+    const { t } = useTranslation(['search'])
+
+    const displayTextShort = t('messages.S20.short')
+    const displayTextLong = t('messages.S20.long') 
+
+    const[message, setMessage] = useState(displayTextLong) //initial state will be the long message
+
+    if (message === displayTextLong){
```
Comment (jlpadilla) [inline frontend/src/routes/SearchPage/components/MessageComponent.tsx:86]: Close, but you need to check that the message id and the display text are the same.
```suggestion
    if (message?[0]?.id === 's20'){
```

Comment (Anxhela21) [reply frontend/src/routes/SearchPage/components/MessageComponent.tsx]: there is an issue with the syntax for grabbing the id dynamically in the translation. 

ex: t(`messages.${messages.data.messages[0].id}.short`)

File: frontend/public/locales/en/search.json [hunk 2]
```diff
@@ -51,5 +51,7 @@
     "search.saved.searches.error": "Query error related to saved search results.",
     "search.saved.searches.section.title": "Saved searches",
     "search.suggested.searches.seaction.title": "Suggested search templates",
-    "searchbar.button.save.search": "Save search"
+    "searchbar.button.save.search": "Save search",
+    "messages.S20.short": "More on disabled clusters",
+    "messages.S20.long": "Currently, search is disabled on some of your managed clusters. Some data might be missing from the console view. See ______ to enable search."
```
Comment (jlpadilla) [inline frontend/public/locales/en/search.json:56]: > "See ______ to enable search."
- We need to add the link to the documentation.
- To move this PR forward, let's omit this sentence.

Comment (Anxhela21) [reply frontend/public/locales/en/search.json:56]: do we have a link to enable search? I thought that was only done through editing the klusterletaddonconfig file.

File: frontend/src/routes/SearchPage/SearchPage.test.tsx [hunk 3]
```diff
@@ -45,6 +50,20 @@ describe('SearchPage', () => {
...
+                    query: GetMessagesDocument,
+                },
+                result: {
+                    data: {
+                        id: 'S20',
+                        kind: 'info',
+                        description: 'Search is disabled on some of your managed clusters.',
+                        __typename: 'Message',
+                    },
+                },
```
Comment (jlpadilla) [inline frontend/src/routes/SearchPage/SearchPage.test.tsx:65]: I expect data to be an array. Need to double check all mocks in this file.

File: frontend/src/console-sdk/console-sdk.ts [hunk 4]
```diff
@@ -1,5 +1,7 @@
 // Copyright (c) 2021 Red Hat, Inc.
 // Copyright Contributors to the Open Cluster Management project
+/* istanbul ignore file */
+// disabling code coverage for this file as it is auto generated
```
Comment (jlpadilla) [inline frontend/src/console-sdk/console-sdk.ts:4]: Remove this.  File is ignored in `sonar-project.properties`
```suggestion
```

File: frontend/src/routes/SearchPage/components/HeaderWithNotification.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+export default function HeaderWithNotification(props: { queryMessages: any[] }) {
+    const { t } = useTranslation(['search'])
+    const messages: any[] = props.queryMessages
+
+    return (
+        <div>
+            <div style={{ outline: 'none', display: 'flex', justifyContent: 'flex-end' }}>
+                <p style={{ flex: 1 }}>
+                    <AcmPageHeader title={t('search')} />
+                </p>
```
Comment (jlpadilla) [inline frontend/src/routes/SearchPage/components/HeaderWithNotification.tsx:19]: Change `<p>` to `<div>` here to resolve these problems in the unit test.
image
```suggestion
                <div style={{ flex: 1 }}>
                    <AcmPageHeader title={t('search')} />
                </div>
```

File: frontend/src/routes/SearchPage/components/HeaderWithNotification.test.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+test('renders clusters disabled message', () => {
+    const disableSearch = [
+        { id: 'S20', kind: 'info', description: 'Search is disabled on some of your managed clusters.' },
+    ]
+    const { getByText } = render(
+        <MockedProvider mocks={[]}>
+            <HeaderWithNotification queryMessages={disableSearch} />
+        </MockedProvider>
+    )
+    expect(getByText).toMatchSnapshot()
```
Comment (jlpadilla) [inline frontend/src/routes/SearchPage/components/HeaderWithNotification.test.tsx:18]: Update tests to validate the rendered element. Will need to update snapshot.
Let's talk an I can explain the reason.
```suggestion
    const { baseElement } = render(<HeaderWithNotification queryMessages={disableSearch} />)
    expect(baseElement).toMatchSnapshot()
```

File: frontend/src/routes/SearchPage/components/HeaderWithNotification.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+import React from 'react'
+import '@patternfly/react-core/dist/styles/base.css'
+import { useTranslation } from 'react-i18next'
+import { AcmPageHeader, AcmInlineStatus, StatusType } from '@open-cluster-management/ui-components'
+import { Card, CardBody } from '@patternfly/react-core'
+
+// 1. Needed to export as default function.
```
Comment (jlpadilla) [inline frontend/src/routes/SearchPage/components/HeaderWithNotification.tsx:9]: These were just to explain why I changed the code.
```suggestion
```

File: frontend/src/routes/SearchPage/components/Modals/SaveAndEditSearchModal.tsx [hunk 8]
```diff
@@ -94,6 +94,12 @@ export const SaveAndEditSearchModal = (props: any) => {
         return state.searchName === '' || (!props.editSearch && props.saveSearch === '')
     }
 
+    // create a warning component for overwriting saved search ( danger label)
+
+    // create a state of the search overwrite warning and set default to blank
+
+    // if saved search name  === previously saved search then set state over search overwrite warning to true
+
```
Comment (jlpadilla) [inline frontend/src/routes/SearchPage/components/Modals/SaveAndEditSearchModal.tsx:102]:
```suggestion
```

**Review by jlpadilla — APPROVED** [review]
/lgtm
Changes look good to merge. Thank you @Anxhela21

**Comment by Anxhela21** [comment]
lgtm

**Review by SherinV — APPROVED** [review]

**Pull request merged**